### PR TITLE
Add seaport v1.4 support

### PR DESCRIPTION
**Pull request by zhongeric**
Forge integration tests will fail until universal router package is published with seaport 1.4 support

### CODE REVIEW

File: src/entities/protocols/seaportV1_4.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import { RoutePlanner, CommandType } from '../../utils/routerCommands'
+import { ConsiderationItem, FulfillmentComponent, Order, SeaportData } from './seaport'
+
+export type AdvancedOrder = Order & {
+  numerator: BigNumber // uint120
+  denominator: BigNumber // uint120
+  extraData: string // bytes
+}
+
+export class SeaportV1_4Trade extends NFTTrade<SeaportData> {
```
Comment (zhongeric) [inline src/entities/protocols/seaportV1_4.ts:15]: Copy and pasted from `./seaport.ts`, subbing out SeaportTrade with SeaportV1_4Trade and changing the CommandType to SeaportV1_4

**Comment by ewilz** [comment]
Obviously can't release this until we get it deployed since we depend on the UNIVERSAL_ROUTER_ADDRESS parameter. So we'll just need to make sure to update that before releasing this version! I'd say put it all in this PR..

File: src/entities/protocols/seaportV1_4.ts (hunk 1 above)
Comment (ewilz) [reply src/entities/protocols/seaportV1_4.ts:15]: any way you can maybe make an abstract class in seaport.ts that both `SeaportTrade` and `SeaportV1_4Trade` both extend so we don't have so much code duplication?

File: src/entities/protocols/seaport.ts [hunk 2]
```diff
@@ -55,13 +61,22 @@ export type AdvancedOrder = Order & {
...
+    } else {
+      super(Market.SeaportV1_4, orders)
+      this.commandType = CommandType.SEAPORT_V1_4
+    }
   }
 
   encode(planner: RoutePlanner, config: TradeConfig): void {
+    const seaportVersion = this.orders[0].version
     for (const order of this.orders) {
+      if (order.version != seaportVersion) throw new Error('Differing Seaport Versions')
```
Comment (ewilz) [inline src/entities/protocols/seaport.ts:79]: why can't we instantiate with different order versions?

File: src/entities/protocols/seaport.ts [hunk 3]
```diff
@@ -92,7 +107,7 @@ export class SeaportTrade extends NFTTrade<SeaportData> {
           100, // TODO: look into making this a better number
         ])
       }
-      planner.addCommand(CommandType.SEAPORT, [this.getTotalPrice().toString(), calldata], config.allowRevert)
+      planner.addCommand(this.commandType, [this.getTotalPrice().toString(), calldata], config.allowRevert)
```
Comment (ewilz) [inline src/entities/protocols/seaport.ts:110]: to add to my last comment: can this just be something like ` planner.addCommand(mapCmd(order.version))`

File: src/entities/protocols/seaport.ts [hunk 4]
```diff
@@ -122,6 +132,15 @@ export class SeaportTrade extends NFTTrade<SeaportData> {
     return totalPrice
   }
 
+  private commandMap(version: SeaportVersion): CommandType {
+    switch (version) {
+      case SeaportVersion.ONE_POINT_ONE:
```
Comment (ewilz) [inline src/entities/protocols/seaport.ts:137]: nit (can merge without!): might be less annoying for integrators to write out `SeaportVersion.V1_1` or `SeaportVersion.V1_4`

**Review by ewilz — APPROVED** [review]
minor nit, not necessary

**Pull request merged**